catalog/plugin/catalog-plugin-resourcestatus/src/test/java/ddf/catalog/plugin/resourcestatus/TestMetacardResourceStatus.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.plugin.resourcestatus;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import ddf.catalog.cache.ResourceCacheInterface;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.plugin.resourcestatus.metacard.MetacardResourceStatus;
+import ddf.catalog.resource.data.ReliableResource;
+
+public class TestMetacardResourceStatus {
+
+    private ResourceCacheInterface cache;
+    private ReliableResource cachedResource;
+    private QueryResponse queryResponse;
+    private MetacardImpl basicMetacard;
+
+    @Before
+    public void setup() {
+        cache = mock(ResourceCacheInterface.class);
+        cachedResource = mock(ReliableResource.class);
+        queryResponse = mock(QueryResponse.class);
+        basicMetacard = basicMetacard();
+    }
+
+    @Test
+    public void testMetacardDisplaysCachedProduct() throws Exception {
+
+        setupCachedMock(basicMetacard);
+
+        MetacardResourceStatus plugin = new MetacardResourceStatus(cache);
+        QueryResponse queryResponse = plugin.process(this.queryResponse);
+
+        Attribute resourceStatusAttribute = getReourceStatusAttribute(queryResponse);
+
+        assertThat(resourceStatusAttribute.getValue(), is(true));
```

Comment:
May want to `verify` that `cache.getValid()` is called with the right parameters.